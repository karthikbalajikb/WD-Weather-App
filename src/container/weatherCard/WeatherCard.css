.paper {
  width: 100% !important;
  margin: auto;
  text-align: center;
  padding-top: 50px;
}
