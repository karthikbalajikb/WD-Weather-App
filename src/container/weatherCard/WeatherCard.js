import React from "react";
import PropTypes from 'prop-types';
import WDTabs from "../../components/tab/TabBar";
import getWeatherDetails from "./WeatherCard.service";
import Typography from 'material-ui/Typography';
import { withStyles } from 'material-ui/styles';

const styles = theme => ({
   paper : {
    width: '100%',
    margin: 'auto',
    textAlign: 'center',
    paddingTop: '50px'
   }
})

class WeatherCard extends React.Component {

    constructor(props) {
        super(props);
        this.state = { report: [] }
    }

    componentDidMount() {
        try {
            if (this.props.selectedCity.length !== 0) {
                getWeatherDetails(this.props.selectedCity).then((resposne) => {
                    this.setState({ report: resposne })
                })
                    .catch((error) => {
                        console.log(error);
                    });
            }
            else {
                this.setState({ report: [] })
            }
        } catch (e) {
            this.setState({ report: [] })
        }
    }

    componentWillReceiveProps(props) {
        try {
            if (this.props.selectedCity.length !== 0) {
                getWeatherDetails(props.selectedCity).then((resposne) => {
                    this.setState({ report: resposne })
                })
                    .catch((error) => {
                        console.log(error);
                    });
            }
            else {
                this.setState({ report: [] })
            }
        } catch (e) {
            this.setState({ report: [] })
        }
    }

    render() {
        const { classes } = this.props;
        return (
            <div>
                {this.state.report.length > 0 ? (
                    <WDTabs weatherData={this.state.report} />
                ) : <Typography variant="headline" className={classes.paper} color="textSecondary">
                        <span> Loading ... </span>
                    </Typography>}
            </div>
        )
    }
}

WeatherCard.propTypes = {
    classes: PropTypes.object.isRequired,
  };
  

export default withStyles(styles)(WeatherCard);