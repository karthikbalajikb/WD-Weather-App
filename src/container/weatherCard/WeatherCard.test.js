import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import WeatherCard from './WeatherCard';
import WDTabs from '../../components/tab/TabBar';

describe('Container: WeatherCard', () => {
  const city = 'Chennai';

  const weatherReport = [
    [
      '31th Jan 2018',
      [
        {
          dateTime: '212324345',
          dt: '09 Feb 2018',
          temp_max: 23,
          description: 'test',
          humidity: 100,
          wind: 20,
          icon: 'ss',
        },
        {
          dateTime: '212324346',
          dt: '10 Feb 2018',
          temp_max: 43,
          description: 'test 222',
          humidity: 90,
          wind: 30,
          icon: 'dd',
        },
      ],
    ],
    [
      '30th Jan 2018',
      [
        {
          dateTime: '2123243458',
          dt: '09 Feb 2018',
          temp_max: 23,
          description: 'test',
          humidity: 100,
          wind: 20,
          icon: 'ss',
        },
        {
          dateTime: '2123243469',
          dt: '10 Feb 2018',
          temp_max: 43,
          description: 'test 222',
          humidity: 90,
          wind: 30,
          icon: 'dd',
        },
      ],
    ],
  ];

  it('Renders without crashing', () => {
    shallow(<WeatherCard selectedCity={city} />);
  });

  it('WeatherCard snapshot test', () => {
    const component = renderer.create(<WeatherCard selectedCity={city} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('report inital state should be empty array', () => {
    const shallowWrap = shallow(<WeatherCard selectedCity={city} />);
    const dropdownState = shallowWrap.state('report');
    expect(dropdownState).toEqual([]);
  });

  describe('Loading state changes', () => {
    it('Loading should be seen when report state is empty', () => {
      const wrap = shallow(<WeatherCard selectedCity={city} />);
      const reportState = wrap.state('report');
      const mountWrap = mount(<WeatherCard selectedCity={city} />);
      const Typography = mountWrap.find('Typography');

      expect(reportState.length).toEqual(0);
      expect(Typography.length).toEqual(1);
      expect(Typography.text()).toEqual(' Loading ... ');
    });

    it('Loading should be removed and WDTabs component must be rendered when report state is not empty', () => {
      const wrap = shallow(<WeatherCard selectedCity={city} />);
      wrap.setState({ report: weatherReport });
      const reportState = wrap.state('report');
      const WDTabsComp = wrap.find(WDTabs);

      expect(reportState.length).toEqual(2);
      expect(WDTabsComp.length).toEqual(1);
    });
  });

  it('Test componentWillReceiveProps() by passing new city', () => {
    const newCity = 'Bangalore';
    const wrap = mount(<WeatherCard selectedCity={city} />);
    wrap.setProps({ selectedCity: newCity });
    const selectedCity = wrap.prop('selectedCity');
    expect(selectedCity).toEqual(newCity);
  });
});
