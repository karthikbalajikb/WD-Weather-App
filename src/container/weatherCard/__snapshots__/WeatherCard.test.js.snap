// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: WeatherCard WeatherCard snapshot test 1`] = `
<div>
  <h1
    className="MuiTypography-root-2 MuiTypography-headline-7 MuiTypography-colorTextSecondary-24 WeatherCard-paper-1"
  >
    <span>
       Loading ... 
    </span>
  </h1>
</div>
`;
